src/states/game/game.gd

```diff
@@ -463,6 +501,15 @@ func instant_defensive_damage() -> void:
 		await wait_for_timer(Global.animation_speed)
 
 
+func battering_ram_aura(square: Vector2i, battering_ram: MeleeUnit) -> void:
+	var unit := get_unit(square)
+	if unit is MeleeUnit and unit.attack_power != 0:
+		if "attack_power" in unit.extra_stats:
+			unit.extra_stats["attack_power"] += int(battering_ram.card_data.extra_data[0])
+		else:
+			unit.extra_stats["attack_power"] = int(battering_ram.card_data.extra_data[0])
+
+
```

Comment:
So I've been thinking about how to clean this up, and I think I've come up with a good solution. Basically we would split up extra stats into two:

Unit Buffs:
- apply to individual units
- stored in `extra_stats`
- always get cleared right before the next turn begins

Tile Buffs:
- Only for battering ram right now
- Stored in the `DropPoint` class, similar to `extra_data` is for Units
- Updated whenever a battering ram is placed (add the buff to the right squares) or moves (subtract the buff from previous squares, add to new ones)

When evaluating the extra stats for attacking or for the tooltip, sum the two together

Does this seem alright?